pkg/client/common.go

```diff
@@ -14,6 +16,21 @@ import (
 	"github.com/buildpacks/pack/pkg/logging"
 )
 
+func (c *Client) addManifestToIndex(ctx context.Context, indexRepoName, repoName string, index imgutil.ImageIndex) error {
+	imageRef, err := name.ParseReference(repoName, name.WeakValidation)
+	if err != nil {
+		return fmt.Errorf("'%s' is not a valid manifest reference: %s", style.Symbol(indexRepoName), err)
+	}
+
+	imageToAdd, err := c.imageFetcher.Fetch(ctx, imageRef.Name(), image.FetchOptions{Daemon: false})
```

Comment:
Are we fetching the whole image here? Or just the metadata?

Reply:
mmm, I think I am fetching the whole image :(

Reply:
Let's check - I think the layers are lazily loaded. We might be okay